src/lancer.ts

```diff
@@ -469,6 +475,17 @@ Hooks.once("init", async function () {
   // NPC components
   Handlebars.registerHelper("tier-selector", npc_tier_selector);
   Handlebars.registerHelper("npc-feat-preview", npc_feature_preview);
+
+  // ------------------------------------------------------------------------
+  // Sliding HUD Zone, including accuracy/difficulty window
+  Hooks.on('renderHeadsUpDisplay', slidingHUD.attach);
+  Hooks.on('targetToken', (_user: User, _token: Token, isNewTarget: boolean) => {
+    macros.refreshTargeting(isNewTarget ? "may open new window" : "only refresh open window");
+  });
+  Hooks.on('createActiveEffect', () => macros.refreshTargeting("only refresh open window"));
+  Hooks.on('deleteActiveEffect', () => macros.refreshTargeting("only refresh open window"));
+  // updateToken triggers on things like token movement (spotter) and probably a lot of other things
+  Hooks.on('updateToken', () => macros.refreshTargeting("only refresh open window"));
```

Comment:
Thoughts on using booleans or constants instead of static strings?

Reply:
Booleans I think are the worst version of this — I mostly don't like how they require intellisense to know what is going on. 

Enums are fine and I'm all for them, but typescript enums are weird and don't actually disappear on compilation — you have to import the value, not the type (see https://stackoverflow.com/questions/68355845/typescript-enums-cannot-be-used-as-a-value-when-imports-using-the-import-type), which means additional `await` steps or wrecking your bundle splitting. 

That's why I feel like using this kind of self-documenting string union type `refreshTargeting(mode: "may open new window" | "only refresh open window")` is, on balance, the best option. As I understand it, static strings like this are interned by javascript runtimes, so it shouldn't be a performance issue.

Reply:
Fair enough! Points 1 & 2 alleviate my concerns; agreed that it essentially acts as an enum without the technical issues.